tests/unit/test_placeholders_with_steps.py

```diff
@@ -214,6 +214,64 @@ def test_map_state_with_placeholders():
     result = Graph(workflow_definition).to_dict()
     assert result == expected_repr
 
+
+def test_map_state_with_placeholders():
+    map_item_value = MapItemValue(schema={
+        'name': str,
+        'age': str
+    })
+
+    map_item_index = MapItemIndex()
+
+    map_state = Map(
+        'MapState01',
+        parameters={
+            "MapIndex": map_item_index,
+            "Name": map_item_value['name'],
+            "Age": map_item_value['age']
+        }
+    )
+    iterator_state = Pass(
+        'TrainIterator',
+        parameters={
+            'ParamA': map_state.output()['X']["Y"]
+        }
+    )
+
+    map_state.attach_iterator(iterator_state)
```

Comment:
question: what happens with this? - does it become part of the workflow definition somewhere? I thought the `Iterator` and `ItemsPath` properties were required for `Map` states

Reply:
ItemsPath is optional. Similar to InputPath and OutputPath, the default is `$`.

Reply:
The iterator is added to the Map State [here](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/src/stepfunctions/steps/states.py#L554-L561) and will be added to the workflow definition when the Map state itself is added to the workflow definition

The [docstring](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/src/stepfunctions/steps/states.py#L543) will need to be updated - it seems like `iterator` is required in the Map State constructor when in fact, it gets creates successfully without it. However, the workflow will fail to create if no iterator is set - we could add a validation [here](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/src/stepfunctions/steps/states.py#L578) and raise an exception if the iterator is not set. At the moment, if fails [here](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/src/stepfunctions/steps/states.py#L735) with the following message
```
ValueError: Expected branch to be a State or a Chain, but got `None`
```

This can be done in another PR to avoid making the scope larger